manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,410 @@
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.gingersnapproject.configuration.Connector;
+import io.gingersnapproject.configuration.EagerRule;
+import io.gingersnapproject.configuration.LazyRule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.LazyCacheRuleSpec;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+/** Creates two informer on k8s configMap for Lazy and Eager Gingersnap rules
+ * ConfigMap names come from properties
+ * gingersnap.k8s.eager-config-map
+ * gingersnap.k8s.lazy-config-map
+*/
+@ApplicationScoped
+public class CacheRuleInformer {
+   @Inject
+   Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   @Inject
+   Event<RuleEvents.LazyRuleAdded> lazyEventAdd;
+   @Inject
+   Event<RuleEvents.LazyRuleRemoved> lazyEventRemove;
+   @Inject
+   Event<RuleEvents.EagerRuleAdded> eagerEventAdd;
+   @Inject
+   Event<RuleEvents.EagerRuleRemoved> eagerEventRemove;
+
+   private SharedIndexInformer<ConfigMap> lazyInformer;
+   private SharedIndexInformer<ConfigMap> eagerInformer;
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleInformer.class);
+
+   
+   void startWatching(@Observes StartupEvent ignore) {
+      log.info("startWatching(): begin");
+      if (client.isUnsatisfied()) {
+         log.info("Kubernetes client not found, not watching config map");
+         return;
+      }
+      try {
+         var cmName = configuration.lazyConfigMapName();
+         if (cmName.isEmpty()) {
+            throw new RuntimeException("Lazy configMap not specified, not watching config map");
+         }
+         var d = new LazyRuleEventDispatcher(lazyEventAdd, lazyEventRemove);
+         lazyInformer = configureInformer(cmName.get(), d);
+         lazyInformer.start();
+      } catch (RuntimeException ex) {
+         log.error("Lazy rules informer startup failed", ex);
+      }
+
+      try {
+         var cmName = configuration.eagerConfigMapName();
+         if (cmName.isEmpty()) {
+            throw new RuntimeException("Eager configMap not specified, not watching config map");
+         }
+         var d = new EagerRuleEventDispatcher(eagerEventAdd, eagerEventRemove);
+         eagerInformer = configureInformer(cmName.get(), d);
+         eagerInformer.start();
+      } catch (RuntimeException ex) {
+         log.error("Eager rules informer startup failed", ex);
+      }
+   }
+
+   public void stop(@Observes ShutdownEvent ignore) {
+      if (lazyInformer != null) {
+         log.info("Shutdown lazyInformer");
+         lazyInformer.close();
+      }
+
+      if (eagerInformer != null) {
+         log.info("Shutdown eagerInformer");
```

Comment:
I think a lot of the logs in this class need to be redefined as  DEBUG messages.